### PR TITLE
SLCORE-312 Add a way to access all the files of a module

### PR DESCRIPTION
**Pull request by damien-urruty-sonarsource**

### CODE REVIEW

File: core/src/test/java/org/sonarsource/sonarlint/core/client/api/common/ClientFileSystemFixtures.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.client.api.common;
+
+import java.util.stream.Stream;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonarsource.sonarlint.core.client.api.common.analysis.ClientInputFile;
+
+public class ClientFileSystemFixtures {
```
Comment (jblievremont) [inline core/src/test/java/org/sonarsource/sonarlint/core/client/api/common/ClientFileSystemFixtures.java:26]: 👍🏻

File: core/src/main/java/org/sonarsource/sonarlint/core/AbstractSonarLintEngine.java [hunk 2]
```diff
@@ -46,4 +53,43 @@ public void fireModuleFileEvent(Object moduleKey, ClientModuleFileEvent event) {
       moduleContainer.getComponentByType(ModuleFileEventNotifier.class).fireModuleFileEvent(event);
     }
   }
+
+  protected <T> T withModule(AbstractAnalysisConfiguration configuration, Function<ComponentContainer, T> consumer) {
```
Comment (jblievremont) [inline core/src/main/java/org/sonarsource/sonarlint/core/AbstractSonarLintEngine.java:57]: Nice! 👍🏻

**Review by jblievremont — APPROVED** [review]
Nice refactoring of the module-related stuff 👍🏻

File: core/src/main/java/org/sonarsource/sonarlint/core/AbstractSonarLintEngine.java [hunk 3]
```diff
@@ -46,4 +53,43 @@ public void fireModuleFileEvent(Object moduleKey, ClientModuleFileEvent event) {
...
+    private final Iterable<ClientInputFile> filesToAnalyze;
+
+    private AnalysisScopeFileSystem(Iterable<ClientInputFile> filesToAnalyze) {
+      this.filesToAnalyze = filesToAnalyze;
+    }
+
+    @Override
+    public Stream<ClientInputFile> files(String suffix, InputFile.Type type) {
+      return files()
+        .filter(file -> file.relativePath().endsWith(suffix))
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/AbstractSonarLintEngine.java:86]: I think it is enough to use filename instead of relativePath

**Review by henryju — APPROVED** [review]

File: core/src/main/java/org/sonarsource/sonarlint/core/AbstractSonarLintEngine.java (hunk 3 above)
Comment (damien-urruty-sonarsource) [reply core/src/main/java/org/sonarsource/sonarlint/core/AbstractSonarLintEngine.java:86]: I didn't find this on ClientInputFile

Comment (henryju) [reply core/src/main/java/org/sonarsource/sonarlint/core/AbstractSonarLintEngine.java:86]: My bad, this is on InputFile

**Pull request merged**